File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        if len(s) <= 1:
            return s
        result = ""
        for i in range(0, len(s)-1):
            odd = self.expand(s, i, i)
            even = self.expand(s, i, i+1)
            if len(odd) > len(result):
                result = odd
            if len(even) > len(result):
                result = even
        return result
    
    def expand(self, s, left, right):
        while left >= 0 and right < len(s) and s[left] == s[right]:
            left -= 1
            right += 1
        return s[left+1 : right]
        

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        if(s.length() <= 1){
            return s;
        }
        String maxSubString = "";
        for(int i = 0; i < s.length() - 1; i++){
            String odd = subStringP(s, i, i);
            String even = subStringP(s, i, i+1);
            if(odd.length() > maxSubString.length()){
                maxSubString = odd;
            }
            if(even.length() > maxSubString.length()){
                maxSubString = even;
            }
        }
        return maxSubString;
    }

    public String subStringP(String s, int left, int right){
        while(left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)){
            left--;
            right++;
        }
        return s.substring(left + 1, right);
    }
}